libyul/ObjectOptimizer.h

```diff
@@ -0,0 +1,98 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#pragma once
+
+#include <libyul/ASTForward.h>
+#include <libyul/Object.h>
+
+#include <liblangutil/EVMVersion.h>
+
+#include <libsolutil/FixedHash.h>
+
+#include <map>
+#include <memory>
+#include <optional>
+
+namespace solidity::yul
+{
+
+enum class Language
+{
+	Yul,
+	Assembly,
+	StrictAssembly,
+};
+
+Dialect const& languageToDialect(Language _language, langutil::EVMVersion _version);
```

Comment:
I still haven't found a good place for this. It really fit the best in `YulStack`. I could create a separate file but that seems like too much boilerplate to me.

Reply:
After removing types and unification of dialects this probably disappears anyways, I guess.

Reply:
True.